ui/src/app/pages/data/concept/concept-add-modal.tsx

```diff
@@ -119,26 +158,32 @@ export const ConceptAddModal = withCurrentWorkspace()
         addedIds: conceptIds
       };
       try {
-        const conceptSet = await conceptSetsApi().updateConceptSetConcepts(
-          namespace, id, selectedSet.id, updateConceptSetReq);
+        const conceptSet = await conceptSetsApi().updateConceptSetConcepts(namespace, id,
+          selectedSet.id, updateConceptSetReq);
         this.setState({saving: false});
         onSave(conceptSet);
       } catch (error) {
         console.error(error);
       }
     } else {
-      const conceptSet: ConceptSet = {
+      let conceptSet: ConceptSet = {
         name: name,
         description: newSetDescription,
         domain: selectedDomain.domain
       };
+      if (this.isSurvey()) {
+        conceptSet = {
+          ...conceptSet,
+          survey: survey
```

Comment:
Could you add this above and keep the const? (Just add to the object `survey: (this.isSurvey() ? survey : null)`?)